fastlane/lib/fastlane/actions/build_app.rb

```diff
@@ -1,13 +1,160 @@
 module Fastlane
   module Actions
-    require 'fastlane/actions/build_ios_app'
-    class BuildAppAction < BuildIosAppAction
-      #####################################################
-      # @!group Documentation
-      #####################################################
+    module SharedValues
+      IPA_OUTPUT_PATH ||= :IPA_OUTPUT_PATH
+      PKG_OUTPUT_PATH ||= :PKG_OUTPUT_PATH
+      DSYM_OUTPUT_PATH ||= :DSYM_OUTPUT_PATH
+      XCODEBUILD_ARCHIVE ||= :XCODEBUILD_ARCHIVE # originally defined in XcodebuildAction
+    end
+
+    class BuildAppAction < Action
+      # rubocop:disable Metrics/PerceivedComplexity
+      def self.run(values)
+        require 'gym'
```

Comment:
why are these required here and not at the top, on-load?

Reply:
All the actions get loaded when fastlane starts and we really only want the action to require things it needs when it runs (otherwise the initial load of fastlane would take longer and load all the things for all the actions)